tikv/pd
# change rule-group delete func to rule-bundle delete func

### What problem does this PR solve?


Issue Number: Close #5967 

### What is changed and how does it work?


- Change `rule-group delete` func to `delRuleBundle`
- Add regex usage message in `rule-group delete` command desc
- Add deprecated message in `rule-group delete` command desc 

### Check List



Tests



- Manual test (add detailed scripts or steps below)
  - Compile & build pd-ctl
  - start a tiup playground, and added few placement rules
  - use `pd-ctl config placement-rules rule-group delete` command to delete one or some groups
  - check and validate the result

Code changes


Side effects

Related changes
- Need to cherry-pick to the release branch

### Release note



```release-note
None.
```
